crates/client/rpc/src/lib.rs

```diff
@@ -531,11 +531,10 @@ where
             .unwrap_or_default();
 
         let block_transactions = block.transactions();
-        let transaction;
         match block_transactions {
             BlockTransactions::Full(transactions) => {
-                transaction = transactions.get(index).ok_or(StarknetRpcApiError::InvalidTxnIndex)?;
-                Ok(transaction.clone())
+                let transaction = transactions.get(index).ok_or(StarknetRpcApiError::InvalidTxnIndex)?;
+                Ok(Transaction::try_from(transaction.clone()).map_err(|e| StarknetRpcApiError::InternalServerError)?)
```

Comment:
Can we double check this is the error returned by pathfinder in such case ? In the long run we should avoid those unexplicit errors which makes it hard to debug. Please add a print too in the meantime.

Reply:
If this fails, it means that internally we are not able to convert a tx from our internal format into its RPC expected format, so I think it's an internal server error.

You mean adding a `println!()` macro in the `map_err` like

```
.map_err(|e| {
    error!("'{e:?}'");
    StarknetRpcApiError::InternalServerError
})
```

Reply:
Error from pathfinder are BlockNotFound and InvalidTxIndex [link](https://github.com/eqlabs/pathfinder/blob/main/crates/rpc/src/v02/method/get_transaction_by_block_id_and_index.rs#LL13C1-L16C3)

Reply:
this error are actually raised before, but here I guess that an internal server error because it's the node not being able to convert from one type to another

Reply:
Let's return Internal server error for now and add the error log @ClementWalter 
then please create an issue so that we dont forget to check if we can have more explicit errors

Reply:
See Issue https://github.com/keep-starknet-strange/madara/issues/457, the fact that we have a `TryFrom` and not a `From` will be solved with this